packages/grafana-data/src/utils/datasource.ts

```diff
@@ -0,0 +1,102 @@
+import { DataSourcePluginOptionsEditorProps, SelectableValue, KeyValue, DataSourceSettings } from '../types';
+
+export const onUpdateDatasourceOption = (props: DataSourcePluginOptionsEditorProps, key: keyof DataSourceSettings) => (
+  event: React.SyntheticEvent<HTMLInputElement | HTMLSelectElement>
```

Comment:
I have small concern about passing the entire event object here. Even though grafana-data has dependency on react types I would like to keep the react dependency as slim as possible. Also this makes the function less testable as you possibly need to pass the entire event object in the test.

I would suggest changing the event to the actual value instead.

Reply:
@dprokop The point of these is that you can use them directly in render without having to create a method for each field. This is why they just wrap a corresponding update function.

Reply:
Yes, I understand this. But it feels wrong with passing the entire event. I would expect it to be used like this in the react component:

`onOptionChange={e => onUpdateDatasourceOption(key, e.currentTarget.value)}`

It is a cosmetic change, but thanks to that we are not introducing another dependency on the react in grafana/data. Also, this becomes more reusable.

Reply:
In that instance, they could just use `updateDatasourcePluginOption` instead?

Reply:
Yes, you are right. Let's leave the definitions as they are then